sklearn/cluster/_birch.py

```diff
@@ -440,6 +441,24 @@ def __init__(self, *, threshold=0.5, branching_factor=50, n_clusters=3,
         self.compute_labels = compute_labels
         self.copy = copy
 
+    # TODO: Remove in 1.2
+    # mypy error: Decorated property not supported
+    @deprecated(  # type: ignore
+        "fit_ is deprecated in 1.0 and will be removed in 1.2"
+    )
+    @property
+    def fit_(self):
+        return self.__fit
```

Comment:
Because `_fit` is already taken.

Reply:
A double underscore in Python does "name mangling" which is not necessarily a thing Python developers are familiar with:

https://en.wikipedia.org/wiki/Name_mangling#Python

I don't think this is necessarily a problem for this code but maybe we should avoid them for the sake of not surprising people?

We could use `_deprecad_fit_flag` instead for instance.

Reply:
Updated PR with a non-named mangled name.